internal/mining/mining_harness_test.go

```diff
@@ -1317,7 +1377,16 @@ func (m *miningHarness) AddFakeUTXO(tx *dcrutil.Tx, blockHeight int64, blockInde
 // new mining harness instance should be created for each test case to ensure
 // that state changes to the underlying fake chain and fake tx source instances
 // do not impact other test cases.
-func newMiningHarness(chainParams *chaincfg.Params) (*miningHarness, []spendableOutput, error) {
+func newMiningHarness(chainParams *chaincfg.Params, chain *fakeChain) (*miningHarness, []spendableOutput, error) {
+	// Set parameters to their default values if nothing passed
+	if chainParams == nil {
+		chainParams = newChainParamsForMiningHarness()
+	}
```

Comment:
An alternative pattern that I like a bit better than passing `nil` for this is defining a separate function.  For example, you could define `newMiningHarnessWithChain` that takes a chain param, and this `newMiningHarness` function could just create the chain and then call `newMiningHarnessWithChain` internally.  Then the caller can decide if they want to use `newMiningHarness` or `newMiningHarnessWithChain`, which is more obvious than knowing you could pass `nil` for the chain.